services/src/main/java/org/keycloak/broker/saml/SAMLIdentityProvider.java

```diff
@@ -186,6 +197,11 @@ public Response keycloakInitiatedBrowserLogout(KeycloakSession session, UserSess
        } else {
             try {
                 LogoutRequestType logoutRequest = buildLogoutRequest(userSession, uriInfo, realm, singleLogoutServiceUrl);
+                if (logoutRequest.getDestination() != null) {
+                    singleLogoutServiceUrl = logoutRequest.getDestination().toString();
+                } else {
+                    return null;
```

Comment:
Same as above

Reply:
Same answer as above. This is inspired from this existing line:

https://github.com/keycloak/keycloak/blob/f8a4f66d6c183de894a8a436f9e35dfa3248e862/services/src/main/java/org/keycloak/broker/saml/SAMLIdentityProvider.java#L183